### ISSUE TITLE
Improve plainbuffer decoding

### ISSUE THREAD

**Issue by xinz** [xinz/ex_aliyun_ots#23]
Here are some benchmark results for note.

```
Operating System: macOS
CPU Information: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
Number of Available Cores: 16
Available memory: 32 GB
Elixir 1.10.3
Erlang 22.3.4.2

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 10 s
memory time: 2 s
parallel: 1
inputs: none specified
Estimated total run time: 42 s

Benchmarking deserialize row binary Commit_a96291c...
Benchmarking deserialize row binary new...
Benchmarking deserialize row binary old...

Name                                            ips        average  deviation         median         99th %
deserialize row binary new                 40393.59      0.0248 ms    ±43.77%      0.0220 ms      0.0680 ms
deserialize row binary Commit_a96291c        721.95        1.39 ms    ±12.75%        1.34 ms        2.10 ms
deserialize row binary old                   437.25        2.29 ms    ±11.38%        2.21 ms        3.25 ms

Comparison:
deserialize row binary new                 40393.59
deserialize row binary Commit_a96291c        721.95 - 55.95x slower +1.36 ms
deserialize row binary old                   437.25 - 92.38x slower +2.26 ms

Memory usage statistics:

Name                                     Memory usage
deserialize row binary new                    8.74 KB
deserialize row binary Commit_a96291c      1319.05 KB - 150.88x memory usage +1310.31 KB
deserialize row binary old                   10.58 KB - 1.21x memory usage +1.84 KB

**All measurements for memory usage were the same**
Operating System: macOS
CPU Information: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
Number of Available Cores: 16
Available memory: 32 GB
Elixir 1.10.3
Erlang 22.3.4.2

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 10 s
memory time: 2 s
parallel: 1
inputs: none specified
Estimated total run time: 42 s

Benchmarking deserialize rows binary Commit_a96291c...
Benchmarking deserialize rows binary new...
Benchmarking deserialize rows binary old...

Name                                             ips        average  deviation         median         99th %
deserialize rows binary new                 34876.97      0.0287 ms    ±38.19%      0.0260 ms      0.0750 ms
deserialize rows binary old                   320.30        3.12 ms    ±11.28%        3.06 ms        4.34 ms
deserialize rows binary Commit_a96291c        298.79        3.35 ms    ±10.86%        3.27 ms        4.55 ms

Comparison:
deserialize rows binary new                 34876.97
deserialize rows binary old                   320.30 - 108.89x slower +3.09 ms
deserialize rows binary Commit_a96291c        298.79 - 116.73x slower +3.32 ms

Memory usage statistics:

Name                                           average  deviation         median         99th %
deserialize rows binary new                   13.17 KB     ±0.00%       13.17 KB       13.17 KB
deserialize rows binary old                  222.38 KB     ±0.08%      222.45 KB      222.50 KB
deserialize rows binary Commit_a96291c      2884.14 KB     ±0.00%     2884.14 KB     2884.14 KB

Comparison:
deserialize rows binary new                   13.17 KB
deserialize rows binary old                  222.38 KB - 16.88x memory usage +209.21 KB
deserialize rows binary Commit_a96291c      2884.14 KB - 218.96x memory usage +2870.97 KB
Operating System: macOS
CPU Information: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
Number of Available Cores: 16
Available memory: 32 GB
Elixir 1.10.3
Erlang 22.3.4.2

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 10 s
memory time: 2 s
parallel: 1
inputs: none specified
Estimated total run time: 42 s

Benchmarking serialize put_row Commit_a96291c...
Benchmarking serialize put_row new...
Benchmarking serialize put_row old...

Name                                       ips        average  deviation         median         99th %
serialize put_row Commit_a96291c        449.93        2.22 ms     ±9.40%        2.16 ms        3.09 ms
serialize put_row new                   422.20        2.37 ms    ±12.79%        2.27 ms        3.60 ms
serialize put_row old                   346.91        2.88 ms    ±12.32%        2.77 ms        4.23 ms

Comparison:
serialize put_row Commit_a96291c        449.93
serialize put_row new                   422.20 - 1.07x slower +0.146 ms
serialize put_row old                   346.91 - 1.30x slower +0.66 ms

Memory usage statistics:

Name                                Memory usage
serialize put_row Commit_a96291c        83.62 KB
serialize put_row new                   73.54 KB - 0.88x memory usage -10.07813 KB
serialize put_row old                 1884.48 KB - 22.54x memory usage +1800.87 KB

**All measurements for memory usage were the same**
Operating System: macOS
CPU Information: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
Number of Available Cores: 16
Available memory: 32 GB
Elixir 1.10.3
Erlang 22.3.4.2

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 10 s
memory time: 2 s
parallel: 1
inputs: none specified
Estimated total run time: 42 s

Benchmarking serialize primary keys Commit_a96291c...
Benchmarking serialize primary keys new...
Benchmarking serialize primary keys old...

Name                                            ips        average  deviation         median         99th %
serialize primary keys new                  18.62 K       53.71 μs    ±28.89%       49.98 μs      127.98 μs
serialize primary keys Commit_a96291c       18.35 K       54.50 μs    ±28.88%       50.98 μs      130.98 μs
serialize primary keys old                  17.78 K       56.23 μs    ±26.37%       51.98 μs      130.60 μs

Comparison:
serialize primary keys new                  18.62 K
serialize primary keys Commit_a96291c       18.35 K - 1.01x slower +0.79 μs
serialize primary keys old                  17.78 K - 1.05x slower +2.52 μs

Memory usage statistics:

Name                                     Memory usage
serialize primary keys new                    1.63 KB
serialize primary keys Commit_a96291c         1.98 KB - 1.22x memory usage +0.35 KB
serialize primary keys old                    4.88 KB - 2.99x memory usage +3.25 KB

**All measurements for memory usage were the same**
```

**Comment by xinz** [comment]
related commit https://github.com/xinz/ex_aliyun_ots/commit/f5fd57417a729a5ddccb7fc4dc01cb5becb4c71c

**Comment by xinz** [comment]
released in [0.10.0](https://hex.pm/packages/ex_aliyun_ots/0.10.0)
